corehq/apps/hqcase/tests/test_case_api_bulk_get.py

```diff
@@ -87,6 +87,41 @@ def test_get_single_case(self):
         assert res.status_code == 200
         assert res.json()['case_id'] == self.case_ids[0]
 
+    def test_get_single_case_by_external_id(self):
+        case_id = '11111111-1111-4111-8111-111111111111'
+        external_id = 'ext-get-case'
+        case_block = CaseBlock(
+            case_id=case_id,
+            case_type='player',
+            case_name='Irina Levitina',
+            external_id=external_id,
+            owner_id=self.web_user.get_id,
+            create=True,
+        )
+        _, [case] = submit_case_blocks(case_block.as_text(), domain=self.domain)
+        populate_case_search_index([case])
+        es_doc = case_search_adapter.get(case_id)
+        expected = serialize_es_case(es_doc)
+
+        base_url = reverse('case_api', args=(self.domain,))
+        if not base_url.endswith('/'):
+            base_url += '/'
+        url = f"{base_url}ext/{external_id}/"
```

Comment:
Why not just name the external ID urlpattern and reverse that?  Looks like you could even give it the same name as long as you reverse it with `kwargs` instead of args:
```python
>>> reverse('case_api', kwargs={'domain': 'esoergel', 'external_id': 'abc123'})
'/a/esoergel/api/v0.6/case/ext/abc123/'
>>> reverse('case_api', kwargs={'domain': 'esoergel', 'case_id': 'abc123'})
'/a/esoergel/api/case/v2/abc123'
```
Though that feels more ambiguous - a separate name would be clearer, I think